x/bet/keeper/bet_settle.go

```diff
@@ -69,15 +69,15 @@ func (k Keeper) SettleBet(ctx sdk.Context, bettorAddressStr, betUID string) erro
 	}
 
 	// check if the bet odds is a winner odds or not and set the bet pointer states
-	if err := processBetResultAndStatus(&bet, market); err != nil {
+	if err := bet.SetResult(&market); err != nil {
```

Comment:
doesn't seem like market should be passed by reference

Reply:
Pointers faster and lighter, if we pass it as value, it will be copied in the stack and uses unnecessary memory.